### ISSUE TITLE
Warning is blocking my build

### ISSUE THREAD

**Issue by ken-experiment** [airbnb/sparsam#17]
I have such warnings that make my build fail.

```
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/include/ruby-2.3.0/universal-darwin17/ruby/config.h:368:31: error:
      invalid suffix on literal; C++11 requires a space between literal and identifier
      [-Wreserved-user-defined-literal]
#define RUBY_ARCH "universal-"RUBY_PLATFORM_OS
                              ^

/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/include/ruby-2.3.0/universal-darwin17/ruby/config.h:369:35: error:
      invalid suffix on literal; C++11 requires a space between literal and identifier
      [-Wreserved-user-defined-literal]
#define RUBY_PLATFORM "universal."RUBY_PLATFORM_CPU"-"RUBY_PLATFORM_OS
                                  ^

/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/include/ruby-2.3.0/universal-darwin17/ruby/config.h:369:55: error:
      invalid suffix on literal; C++11 requires a space between literal and identifier
      [-Wreserved-user-defined-literal]
#define RUBY_PLATFORM "universal."RUBY_PLATFORM_CPU"-"RUBY_PLATFORM_OS
```

By using `gcc-7` it would succeed, but with `gcc` it couldn't.

My suggest fix would be to do 
```
-$CXXFLAGS = " -std=c++0x -O3 -ggdb3 -mtune=native -I./third-party/sparsepp "
+$CXXFLAGS = " -std=c++0x -O3 -ggdb3 -mtune=native -Wno-reserved-user-defined-literal -I./third-party/sparsepp "
```

in `ext/extconf.rb`

**Comment by xavi-** [comment]
Taken from a friend: 

**TLDR: don't use the ruby comes with mac.**

Here is the fix for me:

 1. `brew install rbenv`
 2. `rbenv init`
 3. `rbenv install 2.3.1`, this depends on which ruby version you need for your project
 4. `rbenv versions`, list all the version on your machine currently

```
17:14:12: ~/g/thrift[master]$ rbenv versions
* system
  2.3.1 (set by /Users/jg/.rbenv/version)
````
 5. `rbenv global 2.3.1; rbenv shell 2.3.1`, switch to the newly install one.
```
17:14:12: ~/g/thrift[master]$ rbenv versions
  system
* 2.3.1 (set by /Users/jg/.rbenv/version)
````
 6. Reopen your shell and run `gem install sparsam` . It should work now. (if you are using bundle, you may want to reinstall `bundler` because current one is installed by the unhappy ruby version)

**Comment by andyfangdz** [comment]
Closing as we have a solution (I was never able to reproduce it and that's likely due to I have `rbenv` installed already). Thanks @xavi- !
